subworkflows/local/prepare_genome/main.nf

```diff
@@ -206,33 +254,58 @@ workflow PREPARE_GENOME {
         }
 
     def genomegenerate_input = star_index_check.incompatible
-        .mix(star_index_input.no_index)
+        .mix(ch_star_index_input.no_index)
         .combine(ch_fasta)
-        .map { _meta1, _wrong_index, meta2, fasta ->
-            [meta2, fasta]
+        .map { _meta1, _wrong_index, meta2, fasta_ ->
+            [meta2, fasta_]
         }
 
     STAR_GENOMEGENERATE(genomegenerate_input, ch_gtf)
-    ch_versions = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)
 
     star_index_output = STAR_GENOMEGENERATE.out.index
         .mix(star_index_check.compatible)
         .collect()
 
+    ch_versions = ch_versions.mix(BGZIPTABIX_BCFTOOLS_ANNOTATIONS.out.versions)
+    ch_versions = ch_versions.mix(BGZIPTABIX_DBSNP.out.versions)
+    ch_versions = ch_versions.mix(BGZIPTABIX_KNOWN_INDELS.out.versions)
+    ch_versions = ch_versions.mix(GATK4_CREATESEQUENCEDICTIONARY.out.versions)
+    ch_versions = ch_versions.mix(GFFREAD.out.versions)
+    ch_versions = ch_versions.mix(GTF2BED.out.versions)
+    ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)
+    ch_versions = ch_versions.mix(GUNZIP_GTF.out.versions)
+    ch_versions = ch_versions.mix(REMOVE_UNKNOWN_REGIONS.out.versions)
+    ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions)
+    ch_versions = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)
+    ch_versions = ch_versions.mix(STAR_INDEXVERSION.out.versions)
+    ch_versions = ch_versions.mix(TABIX_BCFTOOLS_ANNOTATIONS.out.versions)
+    ch_versions = ch_versions.mix(TABIX_DBSNP.out.versions)
+    ch_versions = ch_versions.mix(TABIX_KNOWN_INDELS.out.versions)
+    ch_versions = ch_versions.mix(UNTAR.out.versions)
```

Comment:
I personally find it quite confusing to have all the versions at the end? It also makes it harder to keep the list up to date IMO

Reply:
if you prefer, fine by me.
I don't think it matters much to be honest, since it's going to disappear all soon with topics

Reply:
Yeah true, just stating my personal opinion, definitely not a hurdle here